src/dcap_provider.cpp

```diff
@@ -473,36 +556,51 @@ static std::string build_pck_cert_url(const sgx_ql_pck_cert_id_t& pck_cert_id)
 
     const std::string pce_id =
         format_as_big_endian_hex_string(pck_cert_id.pce_id);
-
     std::string version = get_collateral_version();
-    std::stringstream pck_cert_url;
-    pck_cert_url << get_base_url();
+    collateral_url.secondary_base_url << get_secondary_url();
+    collateral_url.primary_base_url << get_base_url();
     if (!version.empty())
     {
-        pck_cert_url << '/';
-        pck_cert_url << version;
+        collateral_url.secondary_base_url << '/' << version;
+        collateral_url.primary_base_url << '/' << version;
     }
-    pck_cert_url << '/' << qe_id;
-    pck_cert_url << '/' << cpu_svn;
-    pck_cert_url << '/' << pce_svn;
-    pck_cert_url << '/' << pce_id;
-    pck_cert_url << '?';
+    collateral_url.secondary_base_url << "/pckcert?";
+    collateral_url.secondary_base_url << "qeid=" << qe_id << '&';
+    collateral_url.secondary_base_url << "cpusvn=" << cpu_svn << '&';
+    collateral_url.secondary_base_url << "pcesvn=" << pce_svn << '&';
+    collateral_url.secondary_base_url << "pceid=" << pce_id << '&';
+
+    collateral_url.primary_base_url << "/pckcert?";
+    collateral_url.primary_base_url << "qeId=" << qe_id << '&';
+    collateral_url.primary_base_url << "cpusvn=" << cpu_svn << '&';
+    collateral_url.primary_base_url << "pcesvn=" << pce_svn << '&';
+    collateral_url.primary_base_url << "pceid=" << pce_id << '&';
 
     std::string client_id = get_client_id();
     if (!client_id.empty())
     {
-        pck_cert_url << "clientid=" << client_id << '&';
+        collateral_url.secondary_base_url << "clientid=" << client_id << '&';
     }
-    pck_cert_url << API_VERSION_LEGACY;
-    return pck_cert_url.str();
+    collateral_url.secondary_base_url << API_VERSION_LEGACY;

```

Comment:
Nit - Version Macro to be explicit about version number instead of Legacy

Reply:
done